src/RequestHandler.ts

```diff
@@ -97,34 +98,34 @@ async function handleFetchAsPromise<T extends HttpMethodTemplate>(
   const url = getUrl(opts);
   log(`_${method.toLowerCase()}`, url);
   const reqOpts = requestConfigToFetchConfig(method, opts);
-  const context = { method, url };
-  const errorContext = { errorToThrow: opts.errorWithCleanerStack, context: { method, url } };
+  const errorContext: ErrorContext = { method, url };
+
   let response: Response;
   try {
     response = await fetch(url, reqOpts);
   } catch (e: any) {
     // This error only occurs when there are network errors and such
-    throw opts.errorWithCleanerStack.addContext(context).setCause(e);
+    throw opts.errorWithCleanerStack.addContext(errorContext).setCause(e);
   }
-  errorContext.errorToThrow.statusCode = response.status;
+  opts.errorWithCleanerStack.statusCode = response.status;
   const consoleOnlyHeader = response.headers.get("x-triply-api");
   if (consoleOnlyHeader) {
     throw opts.errorWithCleanerStack
-      .addContext(context)
+      .addContext(errorContext)
       .setCause(
         new Error(
           `You tried connecting TriplyDB-js to a TriplyDB front-end. Please use the URL of the API instead: ${consoleOnlyHeader}`
         )
       );
   }
-
+  errorContext.message = response.statusText;
```

Comment:
I think this should happen after line 110, just after we have a response. There is possibly an error message that we would like to see in line 114.